tests/vmi_configuration_test.go

```diff
@@ -2667,4 +2667,69 @@ var _ = Describe("Configurations", func() {
 			return nil
 		}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred(), "VMI status IP should match VMI Pod IP")
 	})
+
+	Context("Check KVM CPUID advertisement", func() {
+		var vmi *v1.VirtualMachineInstance
+
+		if tests.IsRunningOnKindInfra() {
```

Comment:
Ah this is causing the strange test failure. This block needs to be inside the `BeforeEach`. Otherwise it gets evaluated and executed in an unpredictable manner.

Reply:
Thanks for fix, pushed to my branch now